### PR TITLE
Automatically unarchive channels that have been used before

### PR DESCRIPTION
**Pull request by evnsio**
This is mostly for use in development.  My instance tried to create comms channel #inc-101, but that channel existed already and had been archived, which the app didn't handle.

### CODE REVIEW

File: response/slack/client.py [hunk 1]
```diff
@@ -79,10 +79,20 @@ def get_channel_id(self, name):
 
             for channel in response["channels"]:
                 if channel["name"] == name:
+                    if channel['is_archived'] and auto_unarchive:
+                        self.unarchive_channel(channel['id'])
                     return channel["id"]
 
         raise SlackError(f"Channel '{name}' not found")
 
+    def unarchive_channel(self, channel_id):
```
Comment (milesbxf) [inline response/slack/client.py]: There is already an unarchive_channel function on line 143, which will override this

File: response/slack/client.py [hunk 2]
```diff
@@ -89,14 +89,14 @@ def get_channel_name(self, id_):
                 return None
             raise
 
-    def get_channel_id(self, name):
+    def get_channel_id(self, name, auto_unarchive=False):
         logger.info(f"Getting channel ID for {name}")
         next_cursor = None
 
         while next_cursor != "":
```
Comment (mattrco) [inline response/slack/client.py]:
```suggestion
        while next_cursor:
```

Comment (mattrco) [inline response/slack/client.py]:
```suggestion
        while next_cursor:
```

Comment (mattrco) [reply response/slack/client.py]: (empty string is False-y)

Comment (mattrco) [reply response/slack/client.py]: I'm not sure why this includes a whitespace change.

**Pull request merged**